### PR TITLE
Group related experiments into projects

### PR DESCRIPTION
**Pull request by gcampax**
Previously, models and datasets would be grouped by "experiments", in S3 and in Tensorboard.
At the same time, the "experiment" key is also passed to the Makefile for generating the dataset and evaluating, and over time it grew to be the specific domain within a single project (e.g. "restaurant" vs "linkedin").

This PR introduces an additional level of grouping, the project. The idea is that project is what each of us is working on: "dlgthingtalk", "schemaorg", "multilingual", etc.
You set the project once in the configuration file and you're good to go, while you can set the experiment on the command line because you probably run multiple experiments within the same project.
Ideally, a project corresponds to a folder in genie-workdirs, and a specific set of thingtalk/genie-toolkit/genienlp branches. So in a way, it supersedes the "workdir" config key.

This is an invasive PR because all S3 paths change. I think we should do a round of cleanup in S3 and delete all the old stuff, so this is a good time to merge it. If you don't want to delete things, you can use `aws s3 mv` to rename stuff from the old names to the new.
(I've been running with this for dialogues for a while now)

### CODE REVIEW

File: train.sh [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 parse_args "$0" "experiment dataset model load_from" "$@"
 shift $n
-check_config "IAM_ROLE OWNER DATASET_OWNER IMAGE TRAIN_TASK_NAME"
+check_config "IAM_ROLE OWNER DATASET_OWNER IMAGE PROJECT TRAIN_TASK_NAME"
 
 load_from=$(check_aws $load_from)
```
Comment (sileix) [inline train.sh:11]: can't we have load_from defaults to None?

**Review by sileix — COMMENTED** [review]
Let's review this, and get it merged today. @Mehrad0711 @s-jse

**Review by Mehrad0711 — APPROVED** [review]

File: train.sh (hunk 1 above)
Comment (Mehrad0711) [reply train.sh:11]: yeah. feel free to do that in this PR.

**Review by s-jse — APPROVED** [review]

File: train.sh (hunk 1 above)
Comment (gcampax) [reply train.sh:11]: Separate PR for that because it needs changes to lib.sh to parse arguments with defaults.

**Pull request merged**